### PR TITLE
feat!: Support querying organization custom roles

### PR DESCRIPTION
**Pull request by tomfeigin**
Added support for querying, creating, updating and deleting organization custom roles.

BREAKING CHANGE: `CreateOrUpdateCustomRoleOptions` has been renamed to `CreateOrUpdateCustomRepoRoleOptions` and `roleID` has been changed from type `string` to `int64`.

### CODE REVIEW

File: github/orgs_custom_roles_test.go
Comment (gmlewis) [inline github/orgs_custom_roles_test.go:160]:
```suggestion

	testNewRequestAndDoFailure(t, methodName, client, func() (*Response, error) {
		return client.Organizations.DeleteCustomOrgRole(ctx, "o", "8030")
	})
}
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @tomfeigin !
Just one addition, please, then I think we will be ready for a second LGTM+Approval before merging.

**Comment by gmlewis** [comment]
(Also, there is no need to force push in this repo - we always squash&merge - see CONTRIBUTING.md for details.)

File: github/orgs_custom_roles.go
Comment (gmlewis) [inline github/orgs_custom_roles.go:111]: According to: https://docs.github.com/en/rest/orgs/organization-roles?apiVersion=2022-11-28#update-a-custom-organization-role
`roleID` should be an integer.

```suggestion
func (s *OrganizationsService) UpdateCustomOrgRole(ctx context.Context, org string, roleID int64, opts *CreateOrUpdateOrgRoleOptions) (*CustomOrgRoles, *Response, error) {
```

File: github/orgs_custom_roles.go
Comment (gmlewis) [inline github/orgs_custom_roles.go:134]: Same here: https://docs.github.com/rest/orgs/organization-roles#delete-a-custom-organization-role

```suggestion
func (s *OrganizationsService) DeleteCustomOrgRole(ctx context.Context, org string, roleID int64) (*Response, error) {
```

Comment (tomfeigin) [reply github/orgs_custom_roles.go:134]: I'll change it but it is aligned with `DeleteCustomRepoRole`

Comment (tomfeigin) [reply github/orgs_custom_roles.go]: I prefer to do that in another PR because here I am just introducing new stuff so it won't break existing usage

**Comment by tomfeigin** [comment]
@gmlewis this specific PR is not a breaking a change as it only introduces new API wrappers (I am pretty sure 🙃 ) can you remove the label please?

File: github/orgs_custom_roles.go
Comment (gmlewis) [inline github/orgs_custom_roles.go:180]: @tomfeigin - if I'm not mistaken, this line and line 203 below are both breaking API changes, as code written by a user of this repo that calls one or both of these endpoints will no longer compile correctly without changing their code.  Agreed?

Comment (tomfeigin) [reply github/orgs_custom_roles.go:180]: Oh you are right, should I keep the previous naming then?

Comment (gmlewis) [reply github/orgs_custom_roles.go:180]: Well, honestly I don't have a problem making breaking API changes, which I suppose might be obvious from the current version number of this repo.  😂 

Also, I think it is always a good idea to make things clearer and easier-to-understand for users of this repo... so I think your name change here was perfectly appropriate and that's also why I recommended fixing the bad field type while we are making a breaking API change.

So I'll leave it up to you, but I think the breaking API change is appropriate and am willing to move forward with it.

**Review by gmlewis — APPROVED** [review]
Thank you, @tomfeigin !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by gsaraf — APPROVED** [review]
Looks good to me :)

**Comment by gmlewis** [comment]
Thank you, @gsaraf !
Merging.

**Pull request merged**
